client_tools.go

```diff
@@ -82,17 +108,27 @@ const (
 
 // Operations is a collection of update operations.
 type Operations struct {
-	ops []Op
+	ops []operation
+}
+
+// EncodeMsgpack encodes Operations as an array of operations.
+func (ops *Operations) EncodeMsgpack(enc *msgpack.Encoder) error {
+	return enc.Encode(ops.ops)
 }
 
 // NewOperations returns a new empty collection of update operations.
 func NewOperations() *Operations {
-	ops := new(Operations)
+	ops := &Operations{[]operation{}}
```

Comment:
```suggestion
	return &Operations{[]operation{}}
```

Reply:
Updated (left `return new(Operations)`).

Reply:
Does encoder.Encode(NewOperations()) work fine? (encodes an empty msgpack array).

Reply:
If we want the result of `Encode(NewOperation())` be equal to `Encode([]interface{}{})`, then we need to use `&Operations{[]operation{}}`. Updated.